packages/office-ui-fabric-react/src/components/DetailsList/DetailsRow.tsx

```diff
@@ -184,7 +186,8 @@ export class DetailsRow extends React.Component<IDetailsRowProps, IDetailsRowSta
         data-automationid='DetailsRow'
         style={ { minWidth: viewport ? viewport.width : 0 } }
         aria-selected={ isSelected }
-        >
+        onContextMenu={ (ev) => { this._onItemContextMenu(this.props.item, this.props.itemIndex, ev); } }
```

Comment:
Maybe it is possible to add the handler to the SelectionZone. It looks like there is already some logic there that finds the item and item index from a particular event target.